src/checkers/debianrepochecker.py

```diff
@@ -155,8 +156,12 @@ def _split_packages_from_page(self, packages_page):
         return packages
 
     def _get_package_from_url(self, package_name, root, dist, component, arch):
-        packages_url = DEB_PACKAGES_URL.format(root=root, dist=dist,
-                                               comp=component, arch=arch)
+        if dist[-1:] == '/' and component is None:
```

Comment:
You shouldn't need the `:` after the `-1` here. A more pythonic way for this could be also `dist.endswith('/')` .

Reply:
Also, should we fail when the dist has not slash at the end but the component is still `None`? I am just guessing it is pretty easy to forget the slash at the end, in which case we'd be formatting `packages_url` with a `None` component.

Reply:
Sorted.